script.js

```diff
@@ -0,0 +1,298 @@
+        switch (keyName) {
+          case 'Backspace':
+            this.properties.value = this.properties.value.slice(0, -1);
+            break;
+          case 'Enter':
+            this.properties.value += '\n';
+            break;
+          case 'Caps Lock':
+            key.classList.toggle('keyboard--key-active');
+            this.toggleCapsLock();
+            break;
+          case 'Shift':
+            break;
+          case 'Del':
+            this.properties.value = this.properties.value.slice(1);
+            break;
+          case 'Tab':
+            this.properties.value += '\t';
+            break;
+          case 'Ctrl':
+            break;
+          case 'Alt':
+            break;
+          case 'Win':
+            break;
+          default:
+            this.properties.value += keyName;
+            break;
+        }
+        if (keyName !== 'Caps Lock') { setTimeout(() => { key.classList.remove('keyboard--key-active'); }, 300); }
+        this.triggerEvent('oninput');
+      }
+    });
+  }
+
+  createKeys() {
+    this.hardKeys.forEach((keyName) => {
+      const key = document.createElement('div');
+      key.classList.add('keyboard--key');
+      this.elements.keys.push(key);
+      switch (keyName) {
+        case 'Backspace':
+          key.classList.add('keyboard--key-wide100');
+          break;
+        case 'Enter':
+          key.classList.add('keyboard--key-wide');
+          break;
+        case 'CapsLock':
+          key.classList.add('keyboard--key-wide100');
+          break;
+        case 'ShiftLeft':
+          key.classList.add('keyboard--key-wide100');
+          break;
+        case 'ShiftRight':
+          key.classList.add('keyboard--key-wide');
+          break;
+        case 'Delete':
+          key.classList.add('keyboard--key-wider');
+          break;
+        case 'Tab':
+          key.classList.add('keyboard--key-wider');
+          break;
+        case 'ControlLeft':
+          break;
+        case 'ControlRight':
+          break;
+        case 'Space':
+          key.classList.add('keyboard--key-widest');
+          break;
+        case 'AltLeft':
+          break;
+        case 'AltRight':
+          break;
+        case 'ArrowLeft':
+          break;
+        case 'ArrowRight':
+          break;
+        case 'ArrowUp':
+          break;
+        case 'ArrowDown':
+          break;
+        case 'MetaLeft':
+          break;
+        default:
+          key.classList.add('keyboard--key-forCaps');
+          break;
+      }
+      this.elements.keysContainer.appendChild(key);
+    });
+  }
+
+  triggerEvent(name) {
+    this.eventHandlers[name](this.properties.value);
+  }
+
+  toggleCapsLock() {
+    this.properties.capsLock = !this.properties.capsLock;
+    const currentLayout = this.properties.capsLock ? this.layouts[this.properties.lang].caps
+      : this.layouts[this.properties.lang].nocaps;
+    this.setKeys(currentLayout);
```

Comment:
These lines repeating in `switchLang` method, so it should be moved in a separate method. DRY